GitUI/Editor/FileViewer.cs

```diff
@@ -395,6 +398,55 @@ string GetFileText()
             }
         }
 
+        private Task ShowOrDeferAsync(string fileName, Func<Task> showFunc)
+        {
+            return ShowOrDeferAsync(GetFileLength(), showFunc);
+
+            long GetFileLength()
+            {
+                var file = new FileInfo(fileName);
+
+                if (!file.Exists)
+                {
+                    file = new FileInfo(_fullPathResolver.Resolve(fileName));
+                }
+
+                if (file.Exists)
+                {
+                    return file.Length;
+                }
+
+                return 0;
+            }
+        }
+
+        private Task ShowOrDeferAsync(long contentLength, Func<Task> showFunc)
+        {
+            const long maxLength = 5 * 1024 * 1024;
+
+            if (contentLength > maxLength)
+            {
+                Clear();
+                Refresh();
+                llShowPreview.Text = $"This file is {contentLength / (1024d * 1024):N1} MB. Showing large files can be slow. Click to show anyway.";
```

Comment:
This needs to be a `TranslationString`

Reply:
Well spotted. I'll do that. Will also rename the field to `_NO_TRANSLATE_` given the designer text will overwrite this.

Reply:
Done.